test/kubetest2-capi/deployer/deployer.go

```diff
@@ -0,0 +1,240 @@
+	"github.com/spf13/pflag"
+	kinddeployer "sigs.k8s.io/kubetest2/kubetest2-kind/deployer"
+	"sigs.k8s.io/kubetest2/pkg/process"
+	"sigs.k8s.io/kubetest2/pkg/types"
+)
+
+// Name is the name of the deployer
+const Name = "capi"
+
+// New implements deployer.New for capi
+func New(opts types.Options) (types.Deployer, *pflag.FlagSet) {
+	kind, flags := kinddeployer.New(opts)
+	d := &deployer{
+		kind:          kind,
+		commonOptions: opts,
+	}
+	bindFlags(d, flags)
+	return d, flags
+}
+
+// assert that New implements types.NewDeployer
+var _ types.NewDeployer = New
+
+type deployer struct {
+	// generic parts
+	commonOptions types.Options
+	kind          types.Deployer
+	// capi specific details
+	provider            string
+	initProviders       []string
+	kubernetesVersion   string
+	controlPlaneCount   string
+	workerCount         string
+	flavor              string
+	useExistingCluster  bool
+	cniManifest         string
+	kubecfgPath         string
+	upTimeout           string
+	workloadClusterName string
+}
+
+func (d *deployer) Kubeconfig() (string, error) {
+	if d.kubecfgPath != "" {
+		return d.kubecfgPath, nil
+	}
+
+	tmpdir, err := ioutil.TempDir("", "kubetest2-capi")
+	if err != nil {
+		return "", err
+	}
+	d.kubecfgPath = path.Join(tmpdir, "kubeconfig.yaml")
+	args := []string{
+		"get", "kubeconfig", d.workloadClusterName,
+	}
+	clusterctl := exec.Command("clusterctl", args...)
+	lines, err := clusterctl.Output()
+	if err != nil {
+		return "", err
+	}
+	if err := ioutil.WriteFile(d.kubecfgPath, lines, 0600); err != nil {
+		return "", err
+	}
+
+	return d.kubecfgPath, nil
+}
+
+// helper used to create & bind a flagset to the deployer
+func bindFlags(d *deployer, flags *pflag.FlagSet) {
+	flags.StringVar(
+		&d.provider, "provider", "", "--provider flag for clusterctl",
+	)
+	flags.StringArrayVar(
+		&d.initProviders, "init-providers", []string{}, "additional --provider arguments for init. will be initialized in addition to the provider set with --provider",
+	)
+	flags.StringVar(
+		&d.kubernetesVersion, "kubernetes-version", "", "--kubernetes-version flag for clusterctl",
+	)
+	flags.StringVar(
+		&d.controlPlaneCount, "control-plane-machine-count", "1", "--control-plane-machine-count flag for clusterctl",
+	)
+	flags.StringVar(
+		&d.workerCount, "worker-machine-count", "1", "--worker-machine-count flag for clusterctl",
+	)
+	flags.StringVar(
+		&d.flavor, "flavor", "", "--flavor flag for clusterctl",
+	)
+	flags.BoolVar(
+		&d.useExistingCluster, "use-existing-cluster", false, "use the existing, currently targeted cluster as the management cluster",
+	)
+	flags.StringVar(
+		&d.upTimeout, "up-timeout", "30m", "maximum time allotted for the --up command to complete",
+	)
+	flags.StringVar(
+		&d.cniManifest, "cni-manifest", "", "automatically install this CNI manifest when the cluster becomes available",
+	)
+	flags.StringVar(
+		&d.workloadClusterName, "workload-cluster-name", "capi-workload-cluster", "the workload cluster name",
```

Comment:
Should we drop the `workload-` prefix, which is part of the CAPI jargon. 
Also, the default could be adapted to reflect the purpose of the cluster instead of focusing on the provisioning method (test something)